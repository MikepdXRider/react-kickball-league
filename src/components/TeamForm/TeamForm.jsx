import React from 'react'

export default function TeamForm({edit = false, teamNameStr, setTeamNameStr, teamCityStr, setTeamCityStr, teamStateStr, setTeamStateStr, submitFn}) {
    return (
        <form onSubmit={(e) => submitFn(e)}>
            <fieldset>
                <legend>{edit ? 'Edit Team' : 'Add Team'}</legend>
                <label htmlFor="name">Name: </label>
                <input type="text" id="name" value={teamNameStr} onChange={(e) => setTeamNameStr(e.target.value)} name="team-name"/>

                <label htmlFor="city">City: </label>
                <input type="text" id="city" value={teamCityStr} onChange={(e) => setTeamCityStr(e.target.value)} name='team-city'/>

                <label htmlFor="state">State: </label>
                <input type="text" id="state" value={teamStateStr} onChange={(e) => setTeamStateStr(e.target.value)} name='team-state' />

                <input type="submit" />
            </fieldset>
        </form>
    )
}
