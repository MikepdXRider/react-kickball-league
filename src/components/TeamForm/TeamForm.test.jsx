import { render, screen } from "@testing-library/react"
import TeamForm from "./TeamForm.jsx"


const testProps = {
    edit: false,
    teamNameStr: 'test-name',
    teamCityStr: 'test-city',
    teamStateStr: 'test-state',
    setTeamNameStr: 'test-fn',
    setTeamCityStr: 'test-fn',
    setTeamStateStr: 'test-fn',
    submitFn: 'test-fn'
}

it('renders TeamForm component', () =>{
    const { container } = render(<TeamForm {...testProps} />)
    const text = screen.getByText(/add team/i);

    expect(container).toMatchSnapshot();
    expect(text).toBeInTheDocument();
})