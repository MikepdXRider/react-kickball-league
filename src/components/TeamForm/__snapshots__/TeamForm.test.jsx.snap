// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TeamForm component 1`] = `
<div>
  <form>
    <fieldset>
      <legend>
        Add Team
      </legend>
      <label
        for="name"
      >
        Name: 
      </label>
      <input
        id="name"
        name="team-name"
        required=""
        type="text"
        value="test-name"
      />
      <label
        for="city"
      >
        City: 
      </label>
      <input
        id="city"
        name="team-city"
        required=""
        type="text"
        value="test-city"
      />
      <label
        for="state"
      >
        State: 
      </label>
      <input
        id="state"
        name="team-state"
        required=""
        type="text"
        value="test-state"
      />
      <input
        type="submit"
      />
    </fieldset>
  </form>
</div>
`;
